File: BettingApp/src/main/java/com/betuiasi/server/persistence/model/TicketRequestModel.java
package com.betuiasi.server.persistence.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class TicketRequestModel {
    @JsonProperty("userUid")
    String userUid;
    @JsonProperty("matches")
    List<SelectedMatch>  matches;
    @JsonProperty("bet")
    Double bet;
    @JsonProperty("potentialWin")
    Double potentialWin;
}

File: BettingApp/src/main/resources/frontend/angular-app/src/app/bilet/models/TicketRequestModel.ts
import {SelectedMatch} from "./SelectedMatch";

export class TicketRequestModel {
  userUid: string;
  matches: SelectedMatch[];
  bet: number;
  potentialWin: number;

  constructor(userUid: string, matches: SelectedMatch[], bet: number, potentialWin: number) {
    this.userUid = userUid;
    this.matches = matches;
    this.bet = bet;
    this.potentialWin = potentialWin;
  }
}
